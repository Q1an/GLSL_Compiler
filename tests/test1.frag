in int a;

// This is a single line comment and should be ignored by the scanner.
void main() {
   int b;
   int d;
   int _underscore;
   int one;  // This is also a singleline comment

   d = 2 + 3 * 4 - 6;
   b = 3;
   a = b + 2;

}

