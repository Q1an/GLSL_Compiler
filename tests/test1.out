in           line 1 cols 1-2 is T_In 
int          line 1 cols 4-6 is T_Int 
a            line 1 cols 8-8 is T_Identifier 
;            line 1 cols 9-9 is ';' 
void         line 4 cols 1-4 is T_Void 
main         line 4 cols 6-9 is T_Identifier 
(            line 4 cols 10-10 is '(' 
)            line 4 cols 11-11 is ')' 
{            line 4 cols 13-13 is '{' 
int          line 5 cols 4-6 is T_Int 
b            line 5 cols 8-8 is T_Identifier 
;            line 5 cols 9-9 is ';' 
int          line 6 cols 4-6 is T_Int 
d            line 6 cols 8-8 is T_Identifier 
;            line 6 cols 9-9 is ';' 
int          line 7 cols 4-6 is T_Int 
_underscore  line 7 cols 8-18 is T_Identifier 
;            line 7 cols 19-19 is ';' 
int          line 8 cols 4-6 is T_Int 
one          line 8 cols 8-10 is T_Identifier 
;            line 8 cols 11-11 is ';' 
d            line 10 cols 4-4 is T_Identifier 
=            line 10 cols 6-6 is '=' 
2            line 10 cols 8-8 is T_IntConstant (value = 2)
+            line 10 cols 10-10 is '+' 
3            line 10 cols 12-12 is T_IntConstant (value = 3)
*            line 10 cols 14-14 is '*' 
4            line 10 cols 16-16 is T_IntConstant (value = 4)
-            line 10 cols 18-18 is '-' 
6            line 10 cols 20-20 is T_IntConstant (value = 6)
;            line 10 cols 21-21 is ';' 
b            line 11 cols 4-4 is T_Identifier 
=            line 11 cols 6-6 is '=' 
3            line 11 cols 8-8 is T_IntConstant (value = 3)
;            line 11 cols 9-9 is ';' 
a            line 12 cols 4-4 is T_Identifier 
=            line 12 cols 6-6 is '=' 
b            line 12 cols 8-8 is T_Identifier 
+            line 12 cols 10-10 is '+' 
2            line 12 cols 12-12 is T_IntConstant (value = 2)
;            line 12 cols 13-13 is ';' 
}            line 14 cols 1-1 is '}' 
